framework/test/gtest/BundleHooksTest.cpp

```diff
@@ -242,53 +267,36 @@ void TestFindHookFailure(const Framework& framework)
   auto fwkListenerToken = framework.GetBundleContext().AddFrameworkListener(
     std::bind(&TestFrameworkListener::Event, &listener, std::placeholders::_1));
 
-  auto bundleA =
-    testing::InstallLib(framework.GetBundleContext(), "TestBundleA");
-  US_TEST_CONDITION_REQUIRED(bundleA, "Non-null bundle");
+  auto bundleA = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+  ASSERT_TRUE(bundleA);
   bundleA.Start();
 
+  framework.GetBundleContext().GetBundle(bundleA.GetBundleId());
+
   // bundle starting and bundle started events
-  US_TEST_CONDITION_REQUIRED(listener.events.size() == 1,
-                             "Test for expected number of framework events");
-
-  std::for_each(
-    listener.events.begin(),
-    listener.events.end(),
-    [](const FrameworkEvent& evt) {
-      US_TEST_CONDITION_REQUIRED(evt.GetThrowable() != nullptr,
-                                 "Test for the existence of an exception");
-      US_TEST_CONDITION_REQUIRED(evt.GetType() ==
-                                   FrameworkEvent::Type::FRAMEWORK_WARNING,
-                                 "Test for the correct framework event type");
-      std::string msg(evt.GetMessage());
-      US_TEST_CONDITION_REQUIRED(
-        std::string::npos != msg.find("Failed to call Bundle FindHook  #"),
-        "Test for the correct event message");
-    });
+  //Test for expected number of framework events
+#ifdef US_BUILD_SHARED_LIBS
+  ASSERT_EQ(listener.events.size(), 1);
+#else
+  ASSERT_GE(static_cast<int>(listener.events.size()), 1);
```

Comment:
what is the # of framework events returned under a static build configuration?

Reply:
for static build config, it is 2.!